pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -209,6 +265,28 @@ func (c *Reconciler) createBuild(tr *v1alpha1.TaskRun) (*buildv1alpha1.Build, er
 		return nil, fmt.Errorf("Task %s has nil BuildSpec", t.Name)
 	}
 
+	bSpec := t.Spec.BuildSpec.DeepCopy()
+	bSpec.Volumes = appendIfMissing(bSpec.Volumes, corev1.Volume{
```

Comment:
since we're copying the buildspec before modifying it, we should be able to append the volume every time? it should never be present

Reply:
removed appendIfMissing( fxn and logic -> append( now